src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2950,6 +2952,16 @@ def openshift_create(cmd, client, resource_group_name, name,  # pylint: disable=
                 namespace='Microsoft.Network', type='virtualNetwork',
                 name=vnet_peer
             )
+    if workspace_resource_id is not None:
+        workspace_resource_id = workspace_resource_id.strip()
+        if not workspace_resource_id.startswith('/'):
+            workspace_resource_id = '/' + workspace_resource_id
+        if workspace_resource_id.endswith('/'):
+            workspace_resource_id = workspace_resource_id.rstrip('/')
+        monitor_profile = OpenShiftManagedClusterMonitorProfile(enabled=True, workspace_resource_id=workspace_resource_id)
```

Comment:
We should also check the provided workspace_resource_id is valid and exists.  I have plan to do the same for AKS as well.

Reply:
Will the RP be doing this server side @JackQuincy ?

Reply:
Yes we are already checking the workspace is valid

Reply:
@jim-minter , doing at client side would help user not running into cluster creation failure because of this. This is we learned from AKS experience.